### ISSUE TITLE
LetsEncrypt Mail Cert Problems after upgrade to 1.0.2

### ISSUE THREAD

**Issue by johnnyq** [hestiacp/hestiacp#509]
### In a few words, please describe the issue that you're experiencing:
I noticed mail SSL Letsencrypt are not working after upgrade to 1.0.2. The certificate shows when I click edit mail domain but the connection via smtp states "failed to verify host Certificate". I created a new mail domain and it worked. I attempted to reissue the LetEncrypt cerificate under mail on a domain by unchecking enable ssl for this domain then clicking save and going back and selecting Enable SSL for this domain and select "Use Lets Encrypt to obtain SSL certificate", but I get the following error when I click save 
Field "ssl certificate, ssl key" can not be blank.

Is there a way I can reissue letsecrypt for each mail doman alternatively via the command line unti this issue gets fixed?

### What steps did you take when the issue occured? 
1. Upgrade from 1.0.1 to 1.0.2
2. Click on Mail Tab
3. Click a domain that aleady has a lets encrypt
4. Uncheck Enable SSL for this domain
5. Click Save
6. Check Enable SSL for this domain
7. Check Use Lets Encrypt to obtain SSL certificate
8. Click Save

### Expected behavior:
It should reissue LE for the selected mail domain

### Operating system distribution and release:
Ubuntu 18.04.2 LXC/Proxmox

### Which version of Hestia Control Panel is currently installed?
1.0.2

### Which branch are you using?
Release

### Which of the following web server configurations are you using?
- Apache + Nginx (default)

### Which of the following server roles are installed on your server?
- DNS
- Mail (Dovecot + Exim)
- MySQL (MariaDB)

**Comment by ScIT-Raphael** [comment]
I can confirm that there seems to be a issue with the exim configuration. Dovecot (pop3/imap) is working properly, exim does send the default hostname, but not the let's encrypt certificate. We will work on a fix asap.

**Comment by johnnyq** [comment]
Awesome! Thanks a bunch!!

**Comment by ioannidesalex** [comment]
@johnnyq can you please try and add this one line according to https://github.com/hestiacp/hestiacp/commit/22883b1aa8b1d4bc5be04b62ade94c3a98adf14d in your /usr/local/hestia/func/main.sh file and let us know if this fixes the issue?

**Comment by johnnyq** [comment]
@ioannidesalex I added the additional line, rebooted and still getting the same error

**Comment by ioannidesalex** [comment]
Thanks @johnnyq for the prompt reply. We will continue working on this.

**Comment by johnnyq** [comment]
Excellent sounds great!

**Comment by johnnyq** [comment]
Anyone know of a quick and dirty fix I only need SSL LE on one host for now.

**Comment by johnnyq** [comment]
Thank for the UI Fix @Lupul it seemed to fix the UI issue i was having but theres still an issue with connecting via smtp as it still says it cant verify host cert.

**Comment by Lupul** [comment]
You can try to remove LE mail domain with this command: (be careful to not reissue too many certs, LE will ratelimit your domain)
```v-delete-letsencrypt-domain my mydomain.tld ' ' yes```

And when you need to remove a mail ssl also uncheck LE checkbox (this is in the PR referenced earlyer #513 )

**Comment by Lupul** [comment]
Can you check the mail cert:
```openssl x509 -in /home/MYUSER/conf/mail/MYDOMAIN.TLD/ssl/MYDOMAIN.TLD.crt -text -noout | egrep "CN|DNS"```

**Comment by johnnyq** [comment]
root@web01:/# openssl x509 -in /home/admin/conf/mail/pittpc.com/ssl/pittpc.com.crt -text -noout | egrep "CN|DNS"
        Issuer: C = US, O = Let's Encrypt, CN = Let's Encrypt Authority X3
        Subject: CN = mail.pittpc.com
                DNS:mail.pittpc.com

**Comment by johnnyq** [comment]
https://www.checktls.com/TestReceiver

if you use this website to check certificate it fails on pittpc.com

with the following error


Cert Hostname DOES NOT VERIFY (mail.pittpc.com != web01.pittpc.com \| DNS:web01.pittpc.com)
--
  |   | So email is encrypted but the host is not verified

**Comment by johnnyq** [comment]
okay I got more info, any new domain I create the Cert Hostname Verifies fine, but any old Domain I had before upgrading to 1.0.2 does not verify even if i uncheck enable ssl in mail and re-enable and reissue LE Certs. I can't just delete the old domain as there are emails accounts connected to it. It seems to be a problem with exim handling Certs for old domains.

**Comment by johnnyq** [comment]
I did some more digging it looks like permssions on the old domain mail conf vs newly created domains Im wondering if this is causing the issue im going to try to match permssions on the one that is working

root@web01:/home/admin/conf/mail/pittpc.com# ls -l ../pittrecycles.com/
total 28
-rw-rw---- 1 Debian-exim mail     0 Jul  8 16:00 aliases
-rw-rw---- 1 Debian-exim mail     0 Jul  8 16:00 antispam
-rw-rw---- 1 Debian-exim mail     0 Jul  8 16:00 antivirus
-rw-r----- 1 root        admin 1403 Jul  8 22:20 apache2.conf
-rw-r----- 1 root        admin 1621 Jul  9 01:01 apache2.ssl.conf
-rw-rw---- 1 Debian-exim mail   887 Jul  8 16:00 dkim.pem
-rw-rw---- 1 Debian-exim mail     0 Jul  8 16:00 fwd_only
-rw-r----- 1 root        admin 1504 Jul  8 22:20 nginx.conf
-rw-r--r-- 1 root        root    45 Jul  9 01:01 nginx.forcessl.conf
-rw-r----- 1 root        admin 1671 Jul  9 01:01 nginx.ssl.conf
-rw-rw---- 1 dovecot     mail     0 Jul  8 16:00 passwd
drwxr-xr-x 2 root        root  4096 Jul  9 01:01 ssl

root@web01:/home/admin/conf/mail/pittpc.com# ls -l
total 40
-rw-rw---- 1 root        mail   867 Jul  8 13:16 aliases
-rw-rw---- 1 root        mail     0 Jul  8 13:16 antispam
-rw-rw---- 1 root        mail     0 Jul  8 13:16 antivirus
-rw-r----- 1 root        admin 1373 Jul  8 13:16 apache2.conf
-rw-r----- 1 root        admin 1573 Jul  9 00:46 apache2.ssl.conf
-rw-rw---- 1 Debian-exim mail   887 Jul  8 13:16 dkim.pem
-rw-rw---- 1 root        mail     8 Jul  8 13:16 fwd_only
-rw-r----- 1 root        admin 1468 Jul  8 13:16 nginx.conf
-rw-r--r-- 1 root        root    45 Jul  9 00:46 nginx.forcessl.conf
-rw-r----- 1 root        admin 1617 Jul  9 00:46 nginx.ssl.conf
-rw-rw---- 1 dovecot     mail   733 Jul  8 13:16 passwd
drw-rw---- 2 root        mail  4096 Jul  9 00:46 ssl

**Comment by johnnyq** [comment]
I solved the issue I changed the permssions on the ssl directory to rwxr-xr-x now exim is reading the SSL Certs. Is theres a script I can run so it sets all directories and files to the right user group and permission?

**Comment by ScIT-Raphael** [comment]
Could you share with us which permission you set where?

**Comment by johnnyq** [comment]
Directory 
/home/admin/conf/mail/pittpc.com/ssl  

Before Permissions
drw-rw---- 2 root mail 4096 Jul 9 00:46 ssl

After Permssions 
drwxr-xr-x 2 root mail 4096 Jul 9 00:46 ssl

**Comment by johnnyq** [comment]
Also if you look at my previous post 4 posts back you will notice many ownership differences between my newly created domain on 1.0.2 and my old domain that was upgraded from 1.0.1. Im wondering if this could be problematic

**Comment by ScIT-Raphael** [comment]
We can confirm that it's a permission issue, we are working on a hotfix release (1.0.3).

**Comment by johnnyq** [comment]
Fantastic!! Thanks! =]

**Comment by ScIT-Raphael** [comment]
We've prepared a new version, would you like to check it on your system?
```
wget https://apt.hestiacp.com/hestia_1.0.3_amd64.deb
dpkg -i hestia_1.0.3_amd64.deb
```

**Comment by ScIT-Raphael** [comment]
The issue has been resolved in hotfix 1.0.3: https://github.com/hestiacp/hestiacp/releases/tag/1.0.3
